lib/jnpr/junos/utils/sw.py

```diff
@@ -170,30 +170,71 @@ def pkgadd(self, remote_package, **kvargs):
         args.update(kvargs)
 
         rsp = self.rpc.request_package_add(**args)
+        return self._parse_pkgadd_response(rsp)
 
+
+        # -------------------------------------------------------------------------
+        # pkgaddNSSU - used to perform NSSU upgrade
+        # -------------------------------------------------------------------------
+
+    def pkgaddNSSU(self, remote_package, **kvargs):
+        """
+        Issue the 'request system software nonstop-upgrade' command on the package.
+
+        :param str remote_package:
+        The file-path to the install package on the remote (Junos) device.
+        """
+
+        rsp = self.rpc.request_package_nonstop_upgrade(
+            package_name=remote_package, **kvargs)
+        return self._parse_pkgadd_response(rsp)
+
+    # -------------------------------------------------------------------------
+    # pkgaddISSU - used to perform ISSU upgrade
+    # -------------------------------------------------------------------------
+
+    def pkgaddISSU(self, remote_package, **kvargs):
+        """
+        Issue the 'request system software nonstop-upgrade' command on the package.
+
+        :param str remote_package:
+          The file-path to the install package on the remote (Junos) device.
+        """
+
+        rsp = self.rpc.request_package_in_service_upgrade(
+            package_name=remote_package, **kvargs)
+        return self._parse_pkgadd_response(rsp)
+
+    def _parse_pkgadd_response(self, rsp):
         got = rsp.getparent()
         rc = int(got.findtext('package-result').strip())
-
-        # return True if rc == 0 else got.findtext('output').strip()
-        return True if rc == 0 else False
+        if rc != 0:
+            self.log("software pkgadd message: %s" % got.findtext('output'))
```

Comment:
Suggest unconditionally logging this message, instead of just logging if there's a failure.